src/Database/Commands/PatchByIndexCommand.ts

```diff
@@ -29,17 +29,16 @@ export class PatchByIndexCommand extends IndexQueryBasedCommand {
   }
 
   public setResponse(response: IResponse): IRavenResponse | IRavenResponse[] | void {
-    const responseBody: IResponseBody = response.body;
-    const status: StatusCode = response.statusCode;
+    const result: IRavenResponse = <IRavenResponse>super.setResponse(response);
 
-    if(!responseBody) {
-      throw new ErrorResponseException(StringUtil.format('Could not find index {0}', this.indexName));
+    if(!response.body) {
+      throw new IndexDoesNotExistException(StringUtil.format('Could not find index {0}', this.indexName));
     }
 
-    if (![StatusCodes.Ok, StatusCodes.Accepted].includes(status)) {
+    if (![StatusCodes.Ok, StatusCodes.Accepted].includes(response.statusCode)) {
```

Comment:
@a-serdukov-mobilunity http://node.green/#ES2016

Looking at Node 6.x the only thing we should avoid using is exponentiation (**) operator. Otherwise I think it's fine to change the target to `es2016`.